src/vos/vos_aggregate.c

```diff
@@ -356,11 +363,18 @@ vos_agg_dkey(daos_handle_t ih, vos_iter_entry_t *entry,
 			rc = vos_obj_iter_pre_aggregate(ih);
 			if (rc < 0)
 				return rc;
-			if (rc != 0) {
+			if (rc == 1) {
 				/** We removed the key so let's reprobe */
 				*acts |= VOS_ITER_CB_DELETE;
 				inc_agg_counter(cont, VOS_ITER_DKEY, AGG_OP_DEL);
 				return 0;
+			} else if (rc == 2) {
+				/** Entry is skipped removed the key so let's reprobe */
```

Comment:
If the aggregation is a full scan for snapshot deletion (or after rebuild done), we can't skip it here.

Reply:
I understand why we can't skip for snapshot deletion.  How do we check for that?   Why rebuild?  If we do rebuild, won't it set the appropriate flags?

Reply:
The need_aggregate() above actually checked this by checking the "VOS_AGG_FL_FORCE_SCAN" flag, I'm wondering if should move these feat flags check into need_aggregate()?

We treat rebuild in a similar way to snapshot, rebuild will create a fence epoch (like snapshot epoch) to avoid aggregation spanning the fence, once rebuild is done, the fence epoch will be deleted and a full scan will be triggered.

Reply:
I wonder whether we should just disable aggregation during rebuild